c++/src/Stream.cc

```diff
@@ -0,0 +1,222 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
We must clearly distinguish the different kinds of `Streams` :). We have ORC Streams and also the in-memory streams. Maybe we can create a new folder `io` and put all the file and memory streams there.